src/pamiq_core/threads/thread_control.py

```diff
@@ -99,26 +103,48 @@ def __init__(self, controller: ThreadController) -> None:
 
 class ControllerCommandHandler:
     """A class, handles commands for thread management, facilitating
-    communication and control between the control thread and other threads.
+    communication and control between the control thread and other threads."""
 
-    Args:
-        controller: The ReadOnlyController object to be read.
-    """
+    def __init__(
+        self,
+        controller: ReadOnlyController,
+        on_paused_callback: OnPausedCallback = lambda: None,
+        on_resumed_callback: OnResumedCallback = lambda: None,
+    ) -> None:
+        """Initialize the ControllerCommandHandler object.
 
-    def __init__(self, controller: ReadOnlyController) -> None:
+        Args:
+            controller: The ReadOnlyController object to be read.
+            on_paused_callback: The callback function to be called when the thread is paused.
+            on_resumed_callback: The callback function to be called when the thread is resumed.
+        """
         self._controller = controller
+        self._on_paused = on_paused_callback
+        self._on_resumed = on_resumed_callback
```

Comment:
```suggestion
        self.on_paused = on_paused_callback
        self.on_resumed = on_resumed_callback
```

コールバック関数の遅延設定を可能にしたいのでこれは publicにしてください。mqttといったライブラリだと割と一般的な書き方で、コールバック関数を後で設定するケースはあります。

```py
handler = ...
handler.on_paused = ... # 遅延設定
```